plugins/meta/bandwidth/main.go

```diff
@@ -17,32 +17,42 @@ package main
 import (
 	"crypto/sha1"
 	"encoding/json"
+	"errors"
 	"fmt"
 
 	"github.com/containernetworking/cni/pkg/skel"
 	"github.com/containernetworking/cni/pkg/types"
 	"github.com/containernetworking/cni/pkg/types/current"
 	"github.com/containernetworking/cni/pkg/version"
-
-	"errors"
 	"github.com/containernetworking/plugins/pkg/ip"
+
 	"github.com/vishvananda/netlink"
 )
 
-type PluginConf struct {
-	types.NetConf
-	RuntimeConfig *struct{} `json:"runtimeConfig"`
-
-	RawPrevResult *map[string]interface{} `json:"prevResult"`
-	PrevResult    *current.Result         `json:"-"`
-
+// BandWidthEntry corresponds to a single entry in the bandwidth argument,
+// see CONVENTIONS.md
+type BandWidthEntry struct {
 	IngressRate  int `json:"ingressRate"`  //Bandwidth rate in Kbps for traffic through container. 0 for no limit. If ingressRate is set, ingressBurst must also be set
 	IngressBurst int `json:"ingressBurst"` //Bandwidth burst in Kb for traffic through container. 0 for no limit. If ingressBurst is set, ingressRate must also be set
 
 	EgressRate  int `json:"egressRate"`  //Bandwidth rate in Kbps for traffic through container. 0 for no limit. If egressRate is set, egressBurst must also be set
 	EgressBurst int `json:"egressBurst"` //Bandwidth burst in Kb for traffic through container. 0 for no limit. If egressBurst is set, egressRate must also be set
 }
 
+type PluginConf struct {
+	types.NetConf
+
+	RuntimeConfig struct {
+		BandWidth *BandWidthEntry `json:"bandWidth,omitempty"`
+	} `json:"runtimeConfig,omitempty"`
+
+	// RuntimeConfig *struct{} `json:"runtimeConfig"`
+
+	RawPrevResult *map[string]interface{} `json:"prevResult"`
+	PrevResult    *current.Result         `json:"-"`
+	BandWidthEntry
```

Comment:
The nested struct can be a pointer, and the json deserializer will handle it correctly - makes it easy to track empty values.

Reply:
We do something similar with the host-local IPAM, if you want to see it in action.

Reply:
Good Point, thanks, Now wit's can be much simpler, PTAL. : )